lib/cc/cli/config_generator.rb

```diff
@@ -80,13 +83,24 @@ def workspace_files
           if non_excluded_paths.empty?
             []
           else
-            find_cmd = "find #{non_excluded_paths.map(&:shellescape).join(' ')} -type f -print0"
-            `#{find_cmd}`.strip.split("\0").map do |path|
-              path.sub(%r{^\.\/}, "")
-            end
+            find_workspace_files
           end
         end
       end
+
+      def find_workspace_files
+        find_cmd = %w[find] + non_excluded_paths + %w[-type f -print0]
+        pid, _, out, err = POSIX::Spawn.popen4(*find_cmd)
+        _, status = Process.waitpid2(pid)
+
+        if status.success?
+          out.read.strip.split("\0").map do |path|
+            path.sub(%r{^\.\/}, "")
+          end
+        else
+          raise FindFailError, "We ran '#{find_cmd.shelljoin}' to find analyzeable files, but the command failed with this error: '#{err.read}'"
```

Comment:
typo: `analyzable`

I think we should decide if this exception is primarily for us or end-users. If it's for us, I'd argue for a more generic construct, something like `CC::Spawn` that takes a command (as Array) and returns `out` or raises a more implementation-specific error:  `raise ProcessError, "command <x, y, z> exited with status <a> and stderr <b>"`. If it's more for the end-users, then I think we shouldn't leak the abstraction so much: `raise ConfigGeneratorError, "Failed to find analyzable files.\n(additional details for us like command run, exit status, etc)"`

WDYT?